islands/VoteBtn.tsx

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type { Item, Vote } from "@/utils/db.ts";
+
+export default function VoteBtn(
+  props: { item: Item; votes: Vote[] | [] },
+) {
+  const upvoted = !!props.votes.filter((item) => item.itemId === props.item.id)
+    .length;
+  const voteId = props.votes.find((item) => item.itemId === props.item.id)?.id;
```

Comment:
To further clarify, I think this island should have an `isVoted` parameter that is determined in the handler. The property could then be fed to a [local Preact signal](https://preactjs.com/guide/v10/signals/#local-state-with-signals), which is then updated according to responses from `/api/vote`.

Reply:
I use `useState` in this case